### PR TITLE
Support Spring native feature for Spring Cloud Azure libraries in 4.0 branch

### PR DESCRIPTION
**Pull request by moarychan**
Make the Spring Cloud Azure modules work with GraalVM using Spring AOT capability, this PR is based on the PR https://github.com/Azure/azure-sdk-for-java/pull/25035.

Related issue:
- https://github.com/Azure/azure-sdk-for-java/issues/20489

Changelog:
- Update module azure-sot-graalvm-support, add storage fetures;
- Update Spring Cloud Azure modules to support Spring AOT feature, it's used to enable Spring Native feature for all Spring Cloud Azure libraries.
- Support Spring Native feature for all Azure Spring Boot samples, https://github.com/Azure-Samples/azure-spring-boot-samples/pull/209
- [WIP]Exclude the module `spring-cloud-azure-native-configuration` building in pipeline job.

Discuss:
- JDK 1.8 compilation failure in Spring CI checking
  Spring Native requires at least JDK11, https://docs.spring.io/spring-native/docs/0.11.1/reference/htmlsingle/#support-language.
- Use a new module called `spring-cloud-azure-native-configuration`  for all modules' native support or save the special configuration in each module package?
  Spring provides the [spring-native-configuration](https://github.com/spring-projects-experimental/spring-native/blob/6cc507a354a37ac59251f51a1aab6a92907ef10f/spring-native-configuration/pom.xml) module to support all Spring related modules.
- `azure-aot-graalvm-support-netty` support will be later

Here is the status of the most recent test：
- Supported Raw SDK starter
  |  Library artifact id |Test passed| Remark  | 
  |---|---|---|
  | spring-cloud-azure-starter-storage-blob  | Yes   |  | 
  | spring-cloud-azure-starter-storage-file-share  | Yes  |   | 
  | spring-cloud-azure-starter-storage-queue  | Yes  |   | 
  |spring-cloud-azure-starter-keyvault-secrets | Yes |  |
  |spring-cloud-azure-starter-keyvault-certificates | Yes|  |
  |spring-cloud-azure-starter-appconfiguration | Yes |  |
  |spring-cloud-azure-starter-cosmos | Failed| SDK Team will release a new version without the Jackson Afterborner module. |
  |spring-cloud-azure-starter-eventhubs| Yes|  |
  |spring-cloud-azure-starter-servicebus| N/A | SDK does not support|

- Supported Spring Data
  |  Library artifact id |Test passed| Remark  | 
  |---|---|---|
  | spring-cloud-azure-starter-data-cosmos  | Failed | SDK Team will release a new version without the Jackson Afterburner module.  | 

- Supported Spring Integration
  |  Library artifact id |Test passed| Remark  | 
  |---|---|---|
  | spring-cloud-azure-starter-integration-storage-queue  | Yes  |   | 
  | spring-cloud-azure-starter-integration-eventhubs  | Yes  |   | 
  | spring-cloud-azure-starter-integration-servicebus  | N/A| SDK does not support  | 

- Supported Spring Cloud Stream
  |  Library artifact id |Test passed| Remark  | 
  |---|---|---|
  | spring-cloud-azure-starter-stream-eventhubs  |Failed |[Spring Cloud Stream is not supported at the comment](https://github.com/spring-projects-experimental/spring-native/issues/1429#issuecomment-1010022171)   | 
  | spring-cloud-azure-starter-stream-servicebus  | N/A| SDK does not support  |

### CODE REVIEW

File: sdk/parents/azure-client-sdk-parent/pom.xml [hunk 1]
```diff
@@ -1375,12 +1391,20 @@
...
             <artifactId>native-maven-plugin</artifactId>
-            <version>0.9.8</version>  <!-- {x-version-update;org.graalvm.buildtools:native-maven-plugin;external_dependency} -->
+            <version>0.9.9</version>  <!-- {x-version-update;org.graalvm.buildtools:native-maven-plugin;external_dependency} -->
+            <extensions>true</extensions>
             <configuration>
               <skip>${graalvm.enable}</skip>
               <buildArgs>--no-fallback</buildArgs>
             </configuration>
             <executions>
+              <execution>
```
Comment (saragluna) [inline sdk/parents/azure-client-sdk-parent/pom.xml:1401]: do we need this?

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/aot/azure-aot-graalvm-support/src/main/java/com/azure/aot/graalvm/support/implementation/features/StorageFileShareFeature.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.aot.graalvm.support.implementation.features;
```
Comment (saragluna) [inline sdk/aot/azure-aot-graalvm-support/src/main/java/com/azure/aot/graalvm/support/implementation/features/StorageFileShareFeature.java:3]: should the package name include spring, like com.azure.spring.xxx

File: sdk/spring/spring-cloud-azure-native-configuration/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/secrets/KeyVaultSecretsHints.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+//
+//@NativeHint(
+//    trigger = SecretClientBuilder.class,
+//    types = @TypeHint(
+//        types = AzureKeyVaultSecretProperties.class,
+//        access = { TypeAccess.DECLARED_CONSTRUCTORS, TypeAccess.DECLARED_METHODS }
+//    )
+//)
+//public class KeyVaultSecretsHints implements NativeConfiguration {
+//}
```
[the remark this answers is not in the document]
Comment (moarychan) [reply sdk/spring/spring-cloud-azure-native-configuration/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/secrets/KeyVaultSecretsHints.java]: Yes, just tested

**Pull request closed without merging**